raft/src/test/java/org/apache/kafka/raft/RaftEventSimulationTest.java

```diff
@@ -758,6 +817,7 @@ void initialize() {
         void poll() {
             try {
                 client.poll();
+                counter.poll(0L);
```

Comment:
Is it ok to have the client and counter on separate threads?

Reply:
The APIs need to be safe for multi-threaded access of course. But yeah, I think for the controller, we want to separate the state machine processing from the IO.